src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/Passes/New/CSharpFormattingPass.cs

```diff
@@ -220,7 +222,30 @@ private async Task<SourceText> FormatCSharpAsync(SourceText generatedCSharpText,
 
         var tree = CSharpSyntaxTree.ParseText(generatedCSharpText, cancellationToken: cancellationToken);
         var csharpRoot = await tree.GetRootAsync(cancellationToken).ConfigureAwait(false);
-        var csharpChanges = RazorCSharpFormattingInteractionService.GetFormattedTextChanges(helper.HostWorkspaceServices, csharpRoot, csharpRoot.FullSpan, options.ToIndentationOptions(), options.CSharpSyntaxFormattingOptions, cancellationToken);
+        var csharpSyntaxFormattingOptions = options.CSharpSyntaxFormattingOptions.AssumeNotNull();
+
+        // Roslyn can be configured to insert a space after a method call, or a dot, but that can break Razor. eg:
```

Comment:
>// Roslyn can be configured to insert a space after a method call, or a dot, but that can break Razor. eg: [](http://example.com/codeflow?start=8&length=106)

Devil's advocate, would it be possible to only augment the C# formatting in @ scenarios instead of @{ scenarios?

Reply:
Possible? Sure, it's just code :)

Off the top of my head I can think of 3 approaches:

1. Add something cool like `#pragma formatter disable SpaceBeforeMethodCall` to Roslyn
2. Use the C# parser to go through the formatted expressions after the fact and remove significant whitespace
3. Do a two-pass format, one with the user options, one for just implicit expressions with these modified options

All 3 sound like they're beyond this PR, and personally I think I'd wait until someone complains about the formatting not honouring their settings before we action it (assuming when they complain, we remember that this quirk exists)